cellmlmanip/mathml2sympy/transpiler.py

```diff
@@ -48,18 +48,12 @@ def transpile(xml_node):
                 print('Hit text node with text "' + text + '"')
         elif child_node.nodeType == child_node.ELEMENT_NODE:
             # Call the appropriate MathML handler function for this tag
-            name = child_node.tagName
-            if name in HANDLERS:
-                # If this tag element itself has children
-                if child_node.childNodes:
-                    # We want to pass the node to the handler, and it will deal with children
-                    sympy_expressions.append(HANDLERS[name](child_node))
-                else:
-                    # This tag has no children
-                    sympy_expressions.append(HANDLERS[name]())
+            tag_name = child_node.tagName
+            if tag_name in HANDLERS:
```

Comment:
This could just be 

```
try:
    sympy_expressions.append(HANDLERS[tag_name](child_node))
catch KeyError:
    raise NotImplemented...
```

(More "pythonic", but also avoids an extra dictionary lookup)

Reply:
That has the danger that if the handler exists but raises `KeyError` for some reason when called, you'll get a misleading error message. `in` is pretty quick for a dict so performance is unlikely to be an issue. If it turns out to be, we could always do something like
```python
handler = HANDLERS.get(tag_name, None)
if handler:
    sympy_expressions.append(handler(child_node))
else:
    raise ...
```